config/confighttp/confighttp.go

```diff
@@ -53,9 +55,13 @@ type HTTPClientSettings struct {
 	// Timeout parameter configures `http.Client.Timeout`.
 	Timeout time.Duration `mapstructure:"timeout"`
 
+	// Deprecated: [v0.67.0] Use OpaqueHeaders instead.
+	Headers map[string]string `mapstructure:"-"`
```

Comment:
Implement "Unmarshal" func on the struct, and during that initialize Headers with the values from `OpaqueHeaders`? That way at least the "unmarshalling" is backwards compatible?

Reply:
> Implement "Unmarshal" func on the struct

Does this work with structs that are not component config structs? I wasn't aware, I can do that.

Note that with the current code unmarshaling is kept backwards compatible in the sense that for a given confmap we end up building the same client as before. But if we care about people accessing the individual fields we should do the unmarshal thing, yes, will fix that

Reply:
Yes works, since 1-2 versions ago :)

Reply:
I think it does not work for embedded structs? See #6668, will open issue if you confirm this is indeed a bug

Reply:
This is not embedded, but that is a bug indeed. Do you plan to fix it?

Reply:
> This is not embedded, but that is a bug indeed

I ran into the issue because the client struct is embedded on the OTLP HTTP exporter https://github.com/open-telemetry/opentelemetry-collector/blob/fb0842a5a200ad1bc7475a4f69baa994310792a4/exporter/otlphttpexporter/config.go#L29 and some tests were failing

> Do you plan to fix it?

Not today, but I can do it next week

Reply:
This no longer blocks this PR, so we can continue the discussion on #6671. It sounds like it's not easy to fix though (may need vendoring the Decode function?)